modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/Checkpointer.java

```diff
@@ -452,9 +456,21 @@ private void syncUpdatedPageStores(
                     return;
                 }
 
-                fsyncDeltaFilePageStoreOnCheckpointThread(entry.getKey(), entry.getValue());
+                GroupPartitionId partitionId = entry.getKey();
+
+                FilePageStore filePageStore = filePageStoreManager.getStore(partitionId.getGroupId(), partitionId.getPartitionId());
+
+                if (filePageStore == null || filePageStore.isMarkedToDestroy()) {
+                    continue;
+                }
+
+                currentCheckpointProgress.onStartPartitionProcessing(partitionId.getGroupId(), partitionId.getPartitionId());
```

Comment:
Looks like all actual I/O made by a `Checkpointer` has to be enclosed in `onStartPartitionProcessing()`/`onFinishPartitionProcessing()` calls. If this is true, I suggest to describe this requirement somewhere (in the class javadoc, or in a comment in the beginning of this class).

Reply:
Changed this code